testsuite/integration-tests/src/test/java/org/jboss/intersmash/testsuite/provision/k8s/HyperfoilKubernetesOperatorProvisionerTest.java

```diff
@@ -0,0 +1,185 @@
+ * limitations under the License.
+ */
+package org.jboss.intersmash.testsuite.provision.k8s;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import org.jboss.intersmash.application.operator.HyperfoilOperatorApplication;
+import org.jboss.intersmash.junit5.IntersmashExtension;
+import org.jboss.intersmash.k8s.KubernetesConfig;
+import org.jboss.intersmash.k8s.client.Kuberneteses;
+import org.jboss.intersmash.provision.k8s.HyperfoilKubernetesOperatorProvisioner;
+import org.jboss.intersmash.provision.olm.OperatorGroup;
+import org.jboss.intersmash.provision.operator.HyperfoilOperatorProvisioner;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.runschema.RunStatisticsWrapper;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.HyperfoilApi;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.invoker.ApiClient;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.invoker.ApiException;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.invoker.Configuration;
+import org.jboss.intersmash.provision.operator.hyperfoil.client.v05.model.Run;
+import org.jboss.intersmash.testsuite.junit5.categories.KubernetesTest;
+import org.jboss.intersmash.testsuite.k8s.NamespaceCreationCapable;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.hyperfoil.v1alpha2.Hyperfoil;
+import io.hyperfoil.v1alpha2.HyperfoilBuilder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@KubernetesTest
+public class HyperfoilKubernetesOperatorProvisionerTest implements NamespaceCreationCapable {
+	private static final Logger logger = LoggerFactory.getLogger(HyperfoilKubernetesOperatorProvisionerTest.class);
+	private static final String NAME = "hyperfoil";
+	private static final HyperfoilKubernetesOperatorProvisioner hyperfoilOperatorProvisioner = initializeOperatorProvisioner();
+
+	private static HyperfoilKubernetesOperatorProvisioner initializeOperatorProvisioner() {
+		HyperfoilKubernetesOperatorProvisioner operatorProvisioner = new HyperfoilKubernetesOperatorProvisioner(
+				new HyperfoilOperatorApplication() {
+					@Override
+					public Hyperfoil getHyperfoil() {
+						Hyperfoil hyperfoil = new HyperfoilBuilder(
+								getName(),
+								// see https://github.com/Hyperfoil/hyperfoil-operator/issues/18, "latest" (default) would fail.
+								"0.24.2").build();
+						return hyperfoil;
+					}
+
+					@Override
+					public String getName() {
+						return NAME;
+					}
+
+					@Override
+					public TargetType getTarget() {
```

Comment:
couldn't this be deduced from execution context? I am assuming we run against K8 OR OCP, not the two at the same time in the same test run

Reply:
Intersmash uses `ProvisionerFactory` instances to dynamically select the relevant provisioner for an Application, i.e. the Application type drives the decision, and this is the logic that we want to build upon.

That being said I changed the implementation in the last push so that there is no longer need for the implementing Application instance to set an enum value, but just to additionally implement either `OpenShiftApplication` or `KubernetesApplication` 

IIUC what you are relating to, is the ability for a test suite that uses Intersmash to execute a set of tests _just_ against K8s or OpenShift in one  single run. This is something that must be implemented in the domain of the client test suite, and not by Intersmash. We actually do something similar in the Intersmash testsuite, where we use the `OpenShiftTest` or `KubernetesTest` categories to select which tests should be run.